File: test-python/leetcode/J101.py
class TreeNode:
    def __init__(self, val=0, left=None, right=None):
        self.val = val
        self.left = left
        self.right = right


class Solution:
    def isSymmetric(self, root: TreeNode) -> bool:
        return self.check(root.left,root.right)
    

    def check(self,q:TreeNode,p:TreeNode) -> bool:
        if not q and not p:
            return True
        if not q or not p:
            return False
        return q.val==p.val and self.check(q.left,p.right) and self.check(q.right,p.left)

File: java-test/java/java-arithmetic/src/main/java/com/leetcode/first/J101.java
package com.leetcode.first;

public class J101 {

    public static void main(String[] args) {

    }
    public boolean isSymmetric(TreeNode root) {
        return check(root.left,root.right);
    }

    public boolean check(TreeNode left,TreeNode right){
        if (left==null && right==null){
            return true;
        }
        if (left==null || right==null){
            return false;
        }

        return left.val== right.val && check(left.left,right.right) && check(left.right,right.left);
    }

    class TreeNode {
        int val;
        TreeNode left;
        TreeNode right;
        TreeNode() {}
        TreeNode(int val) { this.val = val; }
        TreeNode(int val, TreeNode left, TreeNode right) {
            this.val = val;
            this.left = left;
            this.right = right;
        }
    }

}